internal/ipmi/ipmi.go

```diff
@@ -46,14 +49,21 @@ func (i *Ipmi) run(ctx context.Context, command []string) (output string, err er
 		ipmiArgs = append(ipmiArgs, "-H", i.Host)
 	}
 
-	ipmiArgs = append(ipmiArgs, command...)
-	cmd := exec.CommandContext(ctx, i.ipmitool, ipmiArgs...)
-	cmd.Env = []string{fmt.Sprintf("IPMITOOL_PASSWORD=%s", i.Password)}
-	out, err := cmd.CombinedOutput()
-	if ctx.Err() == context.DeadlineExceeded {
-		return string(out), ctx.Err()
-	}
-	return string(out), errors.Wrap(err, strings.TrimSpace(string(out)))
+    for _, cipherString := range ipmiCiphers {
+        ipmiCmd := append(ipmiArgs, cipherString)
+        ipmiCmd = append(ipmiCmd, command...)
+        cmd := exec.CommandContext(ctx, i.ipmitool, ipmiCmd...)
+        cmd.Env = []string{fmt.Sprintf("IPMITOOL_PASSWORD=%s", i.Password)}
+        out, err = cmd.CombinedOutput()
+        if err == nil {
```

Comment:
The err check should include a check for context errors in the loop, 
```go
if err == nil || ctx.Err() != nil {
  break
}
```

Reply:
Amended